pkg/apis/certificates/validation/validation.go

```diff
@@ -537,3 +537,118 @@ func hasDuplicateUsage(usages []certificates.KeyUsage) bool {
 	}
 	return false
 }
+
+func requireSignerNameAsPrefix(signerName string) func(name string, prefix bool) []string {
+	return func(name string, prefix bool) []string {
+		if signerName != "" {
+			escaped := strings.ReplaceAll(signerName, ".", "-")
+			escaped = strings.ReplaceAll(escaped, "/", "--")
+			if !strings.HasPrefix(name, escaped) {
+				return []string{
+					fmt.Sprintf("ClusterTrustBundle for signer %s must be named with prefix %s", signerName, escaped),
+				}
+			}
+		}
+
+		return nil
+	}
+}
+
+// ValidateClusterTrustBundle runs all validation checks on bundle.
+func ValidateClusterTrustBundle(bundle *certificates.ClusterTrustBundle) field.ErrorList {
+	var allErrors field.ErrorList
+
+	metaErrors := apivalidation.ValidateObjectMeta(&bundle.ObjectMeta, false, requireSignerNameAsPrefix(bundle.Spec.SignerName), field.NewPath("metadata"))
+	allErrors = append(allErrors, metaErrors...)
+
+	if bundle.Spec.SignerName != "" {
+		signerNameErrors := ValidateSignerName(field.NewPath("spec", "signerName"), bundle.Spec.SignerName)
+		allErrors = append(allErrors, signerNameErrors...)
+	}
+
+	pemErrors := validatePEMTrustAnchors(bundle.Spec.PEMTrustAnchors)
+	allErrors = append(allErrors, pemErrors...)
+
+	return allErrors
+}
+
+// ValidateClusterTrustBundleUpdate runs all update validation checks on an
+// update.
+func ValidateClusterTrustBundleUpdate(oldBundle, newBundle *certificates.ClusterTrustBundle) field.ErrorList {
+	var allErrors field.ErrorList
+	allErrors = append(allErrors, ValidateClusterTrustBundle(newBundle)...)
+	allErrors = append(allErrors, apivalidation.ValidateObjectMetaUpdate(&newBundle.ObjectMeta, &oldBundle.ObjectMeta, field.NewPath("metadata"))...)
+
+	if newBundle.Spec.SignerName != oldBundle.Spec.SignerName {
+		allErrors = append(allErrors, field.Forbidden(field.NewPath("spec", "signerName"), "updates may not change the signer name"))
+	}
+
+	return allErrors
+}
+
+// validatePEMTrustAnchors rejects intra-block headers, blocks
+// that don't parse as X.509 CA certificates, and duplicate trust anchors.  It
+// requires that at least one trust anchor is provided.
+func validatePEMTrustAnchors(in string) field.ErrorList {
+	var allErrors field.ErrorList
+	path := field.NewPath("spec", "pemTrustAnchors")
+
+	blockDedupe := map[string][]int{}
+
+	rest := []byte(in)
+	var b *pem.Block
+	i := -1
+	for {
+		b, rest = pem.Decode(rest)
+		if b == nil {
+			break
+		}
+		i++
+
+		if b.Type != "CERTIFICATE" {
+			allErrors = append(allErrors, field.Invalid(path, "<value omitted>", fmt.Sprintf("entry %d has bad block type: %v", i, b.Type)))
+			continue
+		}
+
+		if len(b.Headers) != 0 {
+			allErrors = append(allErrors, field.Invalid(path, "<value omitted>", fmt.Sprintf("entry %d has PEM block headers", i)))
+			continue
+		}
+
+		cert, err := x509.ParseCertificate(b.Bytes)
```

Comment:
consider how to insulate ourselves from changes (relaxing or tightening) to the golang stdlib ParseCertificate method... at minimum, on update, if the old and new object have identical PEMTrustAnchors, the update should not fail validation (will require a little plumbing from the Validate___Update call... there's a `ValidationOptions { ValidateCertificateContent bool }` pattern you could pass down to the point where we validate this field)

Reply:
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#new-enum-value-in-existing-field has an example of the validation options struct pattern

Reply:
Done.